backend/src/controller/authController.ts

```diff
@@ -98,8 +100,10 @@ export class AuthController extends BaseController {
                 const dbUser: IDBUserModel = new DBUser();
                 dbUser.fromInterface(registerRequest.user);
                 dbUser.hashPassword(registerRequest.user.password);
-                dbUser.save().then((result) => {
-                    res.json(AuthFactory.createRegisterResponse(true, 'user created.', result.toInterface(), result.id));
+                dbUser.save().then((dbUser) => {
+                    return dbUser.toInterface();
```

Comment:
ööhhmm, ich bin mir nicht sicher, aber ich denke das then müsste auf dem dbUser.toInterface() aufgerufen werden nicht auf dem returnwert von save().then(). dito Login

Reply:
Ist so korrekt.
https://javascript.info/promise-chaining
Mann kann das so schreibe, da toInterface() in Promise zurück gibt, werden die Funktionen hinter einander ausgeführt. So wie du das hier angeregt hast kann man das auch schreiben, dann wächst das ganze nach rechts. Lasse ich so sein.